echopype/utils/io.py

```diff
@@ -32,14 +34,59 @@ def get_files_from_dir(folder):
     return [f for f in os.listdir(folder) if os.path.splitext(f)[1] in valid_ext]
 
 
+def get_zarr_compression(var: xr.Variable, compression_settings: dict) -> dict:
+    """Returns the proper zarr compressor for a given variable type"""
+
+    if np.issubdtype(var.dtype, np.floating):
+        return compression_settings["float"]
+    elif np.issubdtype(var.dtype, np.integer):
+        return compression_settings["int"]
+    elif np.issubdtype(var.dtype, np.str_):
+        return compression_settings["string"]
+    elif np.issubdtype(var.dtype, np.datetime64):
+        return compression_settings["time"]
+    else:
+        raise NotImplementedError(f"Zarr Encoding for dtype = {var.dtype} has not been set!")
+
+
+def set_zarr_encodings(ds: xr.Dataset, compression_settings: dict):
+    """
+    Sets all variable encodings based on zarr default values
+    """
+
+    # create zarr specific encoding
+    encoding = dict()
+    for name, val in ds.variables.items():
+
+        val_encoding = val.encoding
+        val_encoding.update(get_zarr_compression(val, compression_settings))
+        encoding[name] = val_encoding
+
+    return encoding
+
+
 def save_file(ds, path, mode, engine, group=None, compression_settings=None):
     """Saves a dataset to netcdf or zarr depending on the engine
     If ``compression_settings`` are set, compress all variables with those settings"""
-    encoding = (
-        {var: compression_settings for var in ds.data_vars}
-        if compression_settings is not None
-        else {}
-    )
+
+    if compression_settings is not None:
+
+        if "float" in compression_settings:  # only zarr has this key in it
+
+            encoding = set_zarr_encodings(ds, compression_settings)
+
+        else:
+
+            # TODO: below is the encoding we were using for netcdf, we need to make
+            #  sure that the encoding is appropriate for all data variables
+            encoding = (
+                {var: compression_settings for var in ds.data_vars}
+                if compression_settings is not None
+                else {}
+            )
```

Comment:
In addition to making sure the encoding is appropriate, I suggest wrapping the setting of encoding for both zarr and netcdf into a single function, something like `set_storage_encodings(ds: xr.Dataset, compression_settings: dict, engine: str)` for readability.

Alternatively, have a similar function as in the case of zarr, say `set_netcdf_encodings` so the structure is parallel.

Reply:
In reviewing other parts of the code I came across `utils/coding.py::set_encodings()` that also handles some default encoding. Since there was some encoding stuff before your changes in this `io.py` module, it is likely that that function was not comprehensive. I think it will be good to consolidate all the encoding-related functions into `utils/coding.py` so that this type of repetition/confusion do not happen in the future.

Reply:
>In addition to making sure the encoding is appropriate, I suggest wrapping the setting of encoding for both zarr and netcdf into a single function, something like set_storage_encodings(ds: xr.Dataset, compression_settings: dict, engine: str) for readability.
> Alternatively, have a similar function as in the case of zarr, say set_netcdf_encodings so the structure is parallel.

That's a good idea, I will go ahead and do that too.

>In reviewing other parts of the code I came across utils/coding.py::set_encodings() that also handles some default encoding. Since there was some encoding stuff before your changes in this io.py module, it is likely that that function was not comprehensive. I think it will be good to consolidate all the encoding-related functions into utils/coding.py so that this type of repetition/confusion do not happen in the future.

I agree that we should move all encoding related items to `utils/coding.py`. I will go ahead and do that. 

From my understanding, `utils/coding.py::set_encodings()` is really only setting the encoding for the time coordinates (i.e. the values in `DEFAULT_TIME_ENCODING`). No other types of encoding are set. Perhaps in a new issue we should consider renaming this function.

Reply:
> From my understanding, utils/coding.py::set_encodings() is really only setting the encoding for the time coordinates (i.e. the values in DEFAULT_TIME_ENCODING). No other types of encoding are set. Perhaps in a new issue we should consider renaming this function.

You are right. Please change the time as you see fit.

Reply:
Ok, I will create an issue for this and we can do a name change at a later time.